templates/librarylink.cpp.in

```diff
@@ -660,15 +660,18 @@ void @ModelName@_spectrum_impl<Solver_type>::calculate_model_observables(
 /******************************************************************/
 
 template <typename Solver_type>
-void @ModelName@_spectrum_impl<Solver_type>::fill_slha_io(@ModelName@_slha_io& slha_io) const
+void @ModelName@_spectrum_impl<Solver_type>::fill_slha_io(@ModelName@_slha_io& slha_io,
+       const Spectrum_generator_settings& settings) const
 {
    const auto& problems = std::get<0>(models).get_problems();
    const auto force_output = std::get<0>(models).do_force_output();
 
    slha_io.set_spinfo(problems);
    if (!problems.have_problem() || force_output) {
       slha_io.set_spectrum(models);
-      slha_io.set_extra(std::get<0>(models), scales, observables);
+      if (settings.get(Spectrum_generator_settings::calculate_observables)) {
```

Comment:
I think here you're not only disabling the observables, but also the output of the 'scales'.